apps/builder/app/builder/features/style-panel/sections/box-shadows/box-shadow-parser.ts

```diff
@@ -0,0 +1,60 @@
+import * as csstree from "css-tree";
+import { List } from "css-tree/utils";
+import type { CssNode } from "css-tree/utils";
+import { parseCssValue } from "../../shared/parse-css-value";
+import type {
+  InvalidValue,
+  LayersValue,
+  UnparsedValue,
+} from "@webstudio-is/css-data";
+
+export const parseBoxShadow = (
+  boxShadow: string
+): LayersValue | InvalidValue => {
+  let tokenStream = boxShadow.trim();
+  tokenStream = tokenStream.endsWith(";")
+    ? tokenStream.slice(0, -1)
+    : tokenStream;
+
+  const cleanupKeywords = ["box-shadow:"];
+
+  for (const cleanupKeyword of cleanupKeywords) {
+    tokenStream = tokenStream.startsWith(cleanupKeyword)
+      ? tokenStream.slice(cleanupKeyword.length).trim()
+      : tokenStream;
+  }
+
+  const ast = csstree.parse(`${tokenStream},`, { context: "value" });
```

Comment:
Why comma?

Reply:
Just testing out stuff.  Will be removing it 👍